modules/hivemq-edge-module-opcua/src/main/java/com/hivemq/edge/adapters/opcua/client/OpcUaEndpointFilter.java

```diff
@@ -66,7 +67,22 @@ public OpcUaEndpointFilter(@NotNull String configPolicyUri, @NotNull OpcUaAdapte
                 return 1;
             }
             return -1 * Integer.compare(policy1.getPriority(), policy2.getPriority());
-        });
+        }).map(endpointDescription -> endpointUpdater(endpointDescription));
+    }
+
+    private EndpointDescription endpointUpdater(EndpointDescription endpoint) {
+        if (adapterConfig.getUpdateEndpoint()) {
+            String[] parts = adapterConfig.getUri().split("://|:|/");
+            if (parts.length == 1) {
+                return EndpointUtil.updateUrl(endpoint, parts[1]);
+            } else if (parts.length > 1) {
+                return EndpointUtil.updateUrl(endpoint, parts[1], Integer.parseInt(parts[2]));
+            } else {
+                return endpoint;
+            }
+        } else {
+            return endpoint;
+        }
```

Comment:
This can be simplified to:

```suggestion
    private EndpointDescription endpointUpdater(EndpointDescription endpoint) {
        if (adapterConfig.getOverrideUri()) {
            return EndpointUtil.updateUrl(endpoint,
                    EndpointUtil.getHost(adapterConfig.getUri()),
                    EndpointUtil.getPort(adapterConfig.getUri()));
        }
        return endpoint;
    }
```

Reply:
added it to my branch